src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -57,6 +62,13 @@ private Elevator() {
     }
 
     public Command moveElevator(RobotState state) {
+        if(state.elevatorPos>getPose()){
+            up=true;
+            rightElevator.getPIDController().setP(Constants.kElevator.kUpP);
+        }else{
+            up=false;
+            rightElevator.getPIDController().setP(Constants.kElevator.kDownP);
+        }
         return run(
```

Comment:
the if statments should be inside the run command, right now you are doing the if statements in the "static" part of the function, this function is simply a getter function that is run at compile time (I think? or on robot startup), the code isinide the run() function is the ONLY code executed at regular intervals